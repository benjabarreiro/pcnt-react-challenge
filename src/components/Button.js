import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  font-size: 16px;
  line-height: 18.4px;
  font-weight: 400;
  border-radius: 50px;
  padding: 15px 21px;
  ${(props) => `
  ${props.disabledStyle && props.isInForm && 'pointer-events: none'};
    background-color: ${props.disabledStyle ? '#ffffff' : '#000000'};
    color: ${props.disabledStyle ? 'rgba(0, 0, 0, 0.5)' : '#ffffff'};
    width: ${props.isInForm ? '315px' : '129px'};
    border: ${!props.isInForm && props.disabledStyle ? '1px solid #585858' : 'none'};
  `}
`;

const Wrapper = styled.div`
  display: inline;
  ${(props) => `
  ${
    props.isInForm &&
    `@media (max-width: 425px) {
    position: absolute;
    bottom: 20px;
  }`
  }
  ${props.disabledStyle && props.isInForm ? 'cursor: not-allowed' : 'cursor: pointer'}
  `}
`;

export const Button = ({ children, disabledStyle, isInForm }) => {
  return (
    <Wrapper disabledStyle={disabledStyle} isInForm={isInForm}>
      <StyledButton disabledStyle={disabledStyle} isInForm={isInForm} type="submit">
        {children}
      </StyledButton>
    </Wrapper>
  );
};
