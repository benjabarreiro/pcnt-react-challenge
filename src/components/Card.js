import styled from 'styled-components';

export const Card = styled.div`
  background-color: #ffffff;
  border-radius: 20px;
  padding: 17px 23px;
  margin-top: 30px;
`;
