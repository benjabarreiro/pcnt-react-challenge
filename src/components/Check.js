import React from 'react';
import { useContext } from 'react';
import { TasksContext } from '../contexts/Tasks';
import CheckIcon from '../icons/check-icon.png';
import styled from 'styled-components';

const Circle = styled.span`
  width: 20px;
  height: 20px;
  border-radius: 50%;
  cursor: pointer;
  ${(props) => `
		background-color: ${props.completed ? '#FF560B' : '#ffffff'};
		border: ${props.completed ? '1px solid #FF560B' : '1px solid #BCBCBC'};	
	`}
`;
const StyledImg = styled.img`
  position: relative;
  top: 2.5px;
  left: 2.5px;
`;

export const Check = ({ id, completed }) => {
  const { completeTask } = useContext(TasksContext);
  return (
    <Circle role="span" onClick={() => completeTask(completed, id)} completed={completed}>
      {completed && <StyledImg role="img" src={CheckIcon} alt="checked" />}
    </Circle>
  );
};
