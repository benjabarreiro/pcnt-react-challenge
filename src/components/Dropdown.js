import React, { useContext } from 'react';
import styled from 'styled-components';
import { DropdownContext } from '../contexts/Dropdown';

const Container = styled.div`
  position: absolute;
  right: 0;
  top: 39px;
  background-color: #ffffff;
  border-radius: 15px;
  box-shadow: 0px 4px 10px 2px rgba(0, 0, 0, 0.15);
  padding: 13px 0 14px;
`;

const StyledUl = styled.ul`
  list-style: none;
`;

const StyledLi = styled.li`
  padding: 0 16px 0 56px;
  line-height: 32px;
  text-align: right;
  cursor: pointer;
  ${(props) => `color: ${props.selected ? '#FF560B' : '#000000'};`}
  &:hover {
    background-color: #e7e7e7;
  }
`;

export const Dropdown = () => {
  const { dropdownList, selectedHandler } = useContext(DropdownContext);

  return (
    <Container>
      <StyledUl>
        {dropdownList.map((x) => (
          <StyledLi key={x.id} selected={x.selected} onClick={() => selectedHandler(x.id)}>
            {x.text}
          </StyledLi>
        ))}
      </StyledUl>
    </Container>
  );
};
