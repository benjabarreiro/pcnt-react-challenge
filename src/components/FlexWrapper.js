import styled from 'styled-components';

export const FlexWrapper = styled.div`
  display: flex;
  flex-direction: ${(props) => props.direction};
  align-items: ${(props) => props.align};
  ${(props) => props.justify && `justify-content: ${props.justify}`};
  ${(props) => props.mb && `margin-bottom: ${props.mb}px;`}
  ${(props) => props.px && `padding: 0 ${props.px}px;`}
`;
