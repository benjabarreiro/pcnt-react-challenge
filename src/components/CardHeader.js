import React, { useState } from 'react';
import { FlexWrapper } from './FlexWrapper';
import { Title } from './Title';
import { Text } from './Text';
import circlePlusFill from '../icons/circle-plus-fill.png';
import arrowsHead from '../icons/arrows-head.png';
import { Modal } from './Modal';
import { Dropdown } from './Dropdown';
import { useContext } from 'react';
import { DropdownContext } from '../contexts/Dropdown';
import styled from 'styled-components';

const CardHeaderWrapper = styled.div`
  position: relative;
`;

const ActionsWrapper = styled.span`
  cursor: pointer;
`;

export const CardHeader = () => {
  const [showModal, setShowModal] = useState(false);

  const { selectedItem, open, dropdownHandler } = useContext(DropdownContext);

  const closeModal = () => setShowModal(false);

  return (
    <CardHeaderWrapper>
      <FlexWrapper direction="row" justify="space-between" align="center" mb={25}>
        <FlexWrapper direction="row" align="center">
          <Title fontSize={18} lineHeight={22.7} mr={10}>
            To do list
          </Title>
          <ActionsWrapper onClick={() => setShowModal(true)}>
            <img src={circlePlusFill} alt="New list button" />
          </ActionsWrapper>
        </FlexWrapper>
        <ActionsWrapper>
          <FlexWrapper role="span" direction="row" align="center" onClick={dropdownHandler}>
            <Text fontSize={14} lineHeight={16.1} mr={5} fontWeight={400}>
              {selectedItem}
            </Text>
            <img src={arrowsHead} alt="Filter list button" />
          </FlexWrapper>
        </ActionsWrapper>
      </FlexWrapper>
      {open && <Dropdown />}
      {showModal && <Modal closeModal={closeModal} />}
    </CardHeaderWrapper>
  );
};
