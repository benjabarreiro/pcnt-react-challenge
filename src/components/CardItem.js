import React from 'react';
import { Text } from './Text';
import { FlexWrapper } from './FlexWrapper';
import { Check } from './Check';
import { useContext } from 'react';
import { TasksContext } from '../contexts/Tasks';

export const CardItem = ({ title, completed, id }) => {
  const { deleteTask } = useContext(TasksContext);
  return (
    <FlexWrapper direction="row" justify="space-between" align="center" mb={16} px={2}>
      <FlexWrapper direction="row" align="center">
        <Check completed={completed} id={id} />
        <Text
          fontSize={16}
          lineHeight={18.4}
          ml={12}
          fontWeight={400}
          color={completed ? 'rgba(0, 0, 0, 0.5)' : '#000000'}>
          {title}
        </Text>
      </FlexWrapper>
      <Text
        onClick={() => deleteTask(id)}
        fontSize={12}
        lineHeight={30}
        fontWeight={700}
        color="#ffffff"
        colorHover="#5E5E5E"
        underlined={true}>
        Delete
      </Text>
    </FlexWrapper>
  );
};
