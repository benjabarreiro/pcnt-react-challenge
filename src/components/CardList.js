import React, { useContext } from 'react';
import { TasksContext } from '../contexts/Tasks';
import { CardItem } from './CardItem';

export const CardList = () => {
  const { list } = useContext(TasksContext);

  return (
    <div>
      {list.map((x) => (
        <CardItem key={x.id} {...x} />
      ))}
    </div>
  );
};
