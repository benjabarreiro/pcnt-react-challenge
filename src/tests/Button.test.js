import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { Button } from '../components/Button';

describe('Button component', () => {
  it('Button isInForm true available', () => {
    const { Button1 } = render(<Button disabledStyle={true} isInForm={true} />);

    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();

    expect(Button1).toMatchSnapshot();
  });
  it('Button isInForm true disable', () => {
    const { Button2 } = render(<Button disabledStyle={false} isInForm={true} />);

    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();

    expect(Button2).toMatchSnapshot();
  });
  it('Button isInForm false available', () => {
    const { Button3 } = render(<Button disabledStyle={true} />);

    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();

    expect(Button3).toMatchSnapshot();
  });
  it('Button isInForm false disable', () => {
    const { Button4 } = render(<Button disabledStyle={false} />);

    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();

    expect(Button4).toMatchSnapshot();
  });
});
