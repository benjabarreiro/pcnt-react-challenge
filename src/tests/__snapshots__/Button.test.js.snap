// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button isInForm false available 1`] = `undefined`;

exports[`Button component Button isInForm false disable 1`] = `undefined`;

exports[`Button component Button isInForm true available 1`] = `undefined`;

exports[`Button component Button isInForm true disable 1`] = `undefined`;
